src/topology/metric_space/emetric_space.lean

```diff
@@ -98,28 +97,18 @@ class emetric_space (α : Type u) extends has_edist α : Type u :=
 
 /- emetric spaces are less common than metric spaces. Therefore, we work in a dedicated
 namespace, while notions associated to metric spaces are mostly in the root namespace. -/
-variables [emetric_space α]
+variables [pseudoemetric_space α]
 
 @[priority 100] -- see Note [lower instance priority]
-instance emetric_space.to_uniform_space' : uniform_space α :=
-emetric_space.to_uniform_space
+instance pseudoemetric_space.to_uniform_space' : uniform_space α :=
+pseudoemetric_space.to_uniform_space
 
-export emetric_space (edist_self eq_of_edist_eq_zero edist_comm edist_triangle)
+export pseudoemetric_space (edist_self edist_comm edist_triangle)
 
 attribute [simp] edist_self
 
-/-- Characterize the equality of points by the vanishing of their extended distance -/
-@[simp] theorem edist_eq_zero {x y : α} : edist x y = 0 ↔ x = y :=
-iff.intro eq_of_edist_eq_zero (assume : x = y, this ▸ edist_self _)
-
-@[simp] theorem zero_eq_edist {x y : α} : 0 = edist x y ↔ x = y :=
-iff.intro (assume h, eq_of_edist_eq_zero (h.symm))
-          (assume : x = y, this ▸ (edist_self _).symm)
-
-theorem edist_le_zero {x y : α} : (edist x y ≤ 0) ↔ x = y :=
-nonpos_iff_eq_zero.trans edist_eq_zero
-
-@[simp] theorem edist_pos {x y : α} : 0 < edist x y ↔ x ≠ y := by simp [← not_le]
+@[simp] theorem edist_eq_zero_of_self {x y : α} : x = y → edist x y = 0 :=
```

Comment:
Isn't it enough to have a simp lemma for `edist x x = 0`? Because if `h : x = y` is in the local simp set, then it can chain it together with `edist_self` and achieve the same effect as this lemma.

Reply:
I wasn't sure about that, removed.